host-spec/c03-transition.tm

```diff
@@ -157,18 +146,18 @@
   same allocator should be used for any other heap allocation to be used by
   the Polkadot Runtime.
 
-  The size of the provided WASM memory should be based on the
-  <verbatim|:heappages> storage key, where each page has the size of 64KB.
-  This memory shoule be made available to the Polkadot runtime for import
-  under the symbol name <verbatim|memory>.
+  The size of the provided WASM memory should be based on the value of the
+  <verbatim|:heappages> storage key (an unsigned 64-bit integer), where each
+  page has the size of 64KB. This memory shoule be made available to the
+  Polkadot runtime for import under the symbol name <verbatim|memory>.
 
-  <subsubsection|Sending Arguments to Runtime
+  <subsubsection|Sending Data to a Runtime Entry
   ><label|sect-runtime-send-args-to-runtime-enteries>
 
   In general, all data exchanged between the Polkadot Host and the Runtime is
   encoded using SCALE codec described in Section
-  <reference|sect-scale-codec>. As a Wasm function, all runtime entries have
-  the following identical signatures:
+  <reference|sect-scale-codec>. Therefore all runtime entries have the
```

Comment:
> Therefore, all

Might also be useful to say that communication between the Runtime and Host API tries to avoid SCALE as much as possible in order to increase performance.

Reply:
That might be true for the Host API, but given that the Runtime API encodes EVERYTHING passed to and returned from a function, I do no think it is universally true.